### PR TITLE
Recent projects and directories in file selector

### PR DESCRIPTION
**Pull request by marioba**
Add two sections on the root of the file selector with the last 3 used directories and the last 5 used projects.

screenshot_20190305-102213

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-499-45fcfd091c936f5985f973c9f2ab6f216c95dd80-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-499-45fcfd091c936f5985f973c9f2ab6f216c95dd80-x86.apk)

File: android/res/layout/list_separator.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="fill_parent"
+    android:layout_height="wrap_content"
+    android:background="#AAAAAA"
```
Comment (mbernasocchi) [inline android/res/layout/list_separator.xml]: OK, for now, but aren't there standard colors?

File: android/res/layout/list_separator.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <TextView
+        android:id="@+id/separator"
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:gravity="left"
+        android:padding="8dp"
+        android:textSize="18dp"
+        android:text=""
+        android:textAllCaps="false"
+        android:textColor="#FFF"
```
Comment (mbernasocchi) [inline android/res/layout/list_separator.xml]: same as above in regard to standard colors

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-499-80ed87b2c3f167c8b7c09a2d82f6cdc691370ecd-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-499-80ed87b2c3f167c8b7c09a2d82f6cdc691370ecd-x86.apk)

File: android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 3]
```diff
@@ -53,15 +58,43 @@ public void onCreate(Bundle bundle) {
...
                 }
             }
 
             setTitle(getString(R.string.select_project));
+            Collections.sort(values);
+
+            String lastUsedDirs = sharedPreferences.getString("LastUsedDirectories", null);
+            Log.d(TAG, "lastUsedDirs: " + lastUsedDirs);
+            if (lastUsedDirs != null){
+                String[] lastUsedDirsArray = lastUsedDirs.split(";");
```
Comment (mbernasocchi) [inline android/src/ch/opengis/qfield/QFieldProjectActivity.java]: Maybe as mentioned do "--;--"

File: android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 4]
```diff
@@ -53,15 +58,43 @@ public void onCreate(Bundle bundle) {
...
+                    File f = new File(lastUsedDirsArray[i]);
+                    if(f.exists()){
+                        values.add(new QFieldProjectListItem(f, f.getName(), R.drawable.directory, QFieldProjectListItem.TYPE_ITEM));
+                    }
+                }
+            }
+
+            String lastUsedProjects = sharedPreferences.getString("LastUsedProjects", null);
+            if (lastUsedProjects != null){
+                String[] lastUsedProjectsArray = lastUsedProjects.split(";");
```
Comment (mbernasocchi) [inline android/src/ch/opengis/qfield/QFieldProjectActivity.java]: Maybe as mentioned do "--;--"

File: android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 5]
```diff
@@ -113,6 +148,39 @@ protected void onListItemClick(ListView l, View v, int position, long id) {
...
+            lastUsedProjectsArray.remove(file.getPath());
+            if (lastUsedProjectsArray.size() >= 5){
+                lastUsedProjectsArray.remove(0);
+            }
+            // Add the project path to the array
+            lastUsedProjectsArray.add(file.getPath());
+
+            // Write the recent directories and projects into the shared preferences
+            SharedPreferences.Editor editor = sharedPreferences.edit();
+            editor.putString("LastUsedDirectories", TextUtils.join(";", lastUsedDirsArray));
```
Comment (mbernasocchi) [inline android/src/ch/opengis/qfield/QFieldProjectActivity.java]: Maybe as mentioned do "--;--"

File: android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 6]
```diff
@@ -113,6 +148,39 @@ protected void onListItemClick(ListView l, View v, int position, long id) {
...
+            editor.putString("LastUsedProjects", TextUtils.join(";", lastUsedProjectsArray));
```
Comment (mbernasocchi) [inline android/src/ch/opengis/qfield/QFieldProjectActivity.java]: Maybe as mentioned do "--;--"

File: android/res/layout/list_separator.xml (hunk 1 above)
Comment (marioba) [reply android/res/layout/list_separator.xml]: What do you mean?

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-499-6d7732a4c0cb35c776e849f93c71c6ae63afa1e2-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-499-6d7732a4c0cb35c776e849f93c71c6ae63afa1e2-x86.apk)

**Pull request merged**